### PR TITLE
Update KubeVirt to use admissionregistration.k8s.io/v1

### PR DESCRIPTION
**Pull request by acardace**
**What this PR does / why we need it**:
Replaces all occurrences of admissionregistration.k8s.io/v1beta1 with admissionregistration.k8s.io/v1
as the former is deprecated and will be removed from Kubernetes 1.22.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/resource/apply/reconcile.go [hunk 1]
```diff
@@ -336,21 +337,23 @@ func (r *Reconciler) createDummyWebhookValidator() error {
...
-		webhooks = append(webhooks, admissionregistrationv1beta1.ValidatingWebhook{
-			Name:          fmt.Sprintf("%s-tmp-validator", crd.Name),
-			FailurePolicy: &failurePolicy,
-			Rules: []admissionregistrationv1beta1.RuleWithOperations{{
-				Operations: []admissionregistrationv1beta1.OperationType{
-					admissionregistrationv1beta1.Create,
+		webhooks = append(webhooks, admissionregistrationv1.ValidatingWebhook{
+			Name:                    fmt.Sprintf("%s-tmp-validator", crd.Name),
+			AdmissionReviewVersions: []string{"v1", "v1beta1"},
+			SideEffects:             &sideEffectNone,
```
Comment (jean-edouard) [inline pkg/virt-operator/resource/apply/reconcile.go:343]: Non-trivial changes spotted

**Review by jean-edouard — APPROVED** [review]
Thank you!
Probably worthy of a release note.
/ok-to-test
/approve

**Comment by jean-edouard** [comment]
/lgtm cancel
(didn't mean to approve + lgtm)

**Review by EdDev — APPROVED** [review]
Thank you!

File: pkg/virt-operator/resource/apply/reconcile.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-operator/resource/apply/reconcile.go:343]: Yes, the `AdmissionReviewVersions` is now mandatory with v1 `AdmissionReview` but I think that's making the downgrade test fail, I will have to fix that.

Comment (EdDev) [reply pkg/virt-operator/resource/apply/reconcile.go:343]: > Yes, the `AdmissionReviewVersions` field is now mandatory with v1 `AdmissionReview` but I think that's making the downgrade test fail, I will have to fix that.

You mean the tests that fail on the 1.18 and 1.19 jobs?
Do you see it fail locally as well?

**Review by ashleyschuett — APPROVED** [review]

**Comment by xpivarc** [comment]
/hold 
Based on https://github.com/kubevirt/kubevirt/pull/5240#discussion_r596673758
Lets see what @acardace finds and then unhold.

**Comment by acardace** [comment]
I fixed the func test that was failing and added a unit test for Validation/MutationWebhookConfiguration conversion between v1beta1 and v1. /unhold

**Pull request merged**